main.sh

```diff
@@ -55,68 +55,107 @@ if [ "$1" = "list" ]; then
     exit
 fi
 
-use_same=""
-use_auth=""
-echo -e "
-\e[1m\e[33mProxyMan
-=========\e[0m
-Tool to set up system wide proxy settings on Linux.
-\e[2m\e[33m🌟\e[0m\e[3m Star it \e[0m : \e[4m\e[34m https://github.com/himanshub16/ProxyMan \e[0m
+http_host=""
+http_port=""
+use_same="n"
+https_host=""
+https_port=""
+ftp_host=""
+ftp_port=""
+username=""
+password=""
+save_for_reuse=""
+profile_name=""
+use_auth="n"
 
- \e[4mThe following options are available : \e[0m
-\e[1m set \e[0m    : \e[2m Set proxy settings \e[0m
-\e[1m unset \e[0m  : \e[2m Unset proxy settings \e[0m
-\e[1m list \e[0m   : \e[2m List current settings \e[0m
 
-"
-
-read -p " Enter your choice : " choice
+if [ "$1" = "load" ]; then
+    if [ "$2" = "" ]; then
+        echo -ne "\e[1m \e[31mPlease provide a config! \e[0m"
+        echo
+        exit 
+    fi
+    
+    if [ ! -e profiles/"$2".txt ]; then
+        echo -ne "\e[1m \e[31mFile does not exist! \e[0m"
+        echo
+        exit
+    fi
 
-if [[ (! "$choice" = "set") && (! "$choice" = "unset") && (! "$choice" = "list") ]]; then
-    echo "Invalid choice! Start again."
+    mapfile -t loaded_config < profiles/"$2".txt
+        http_host=${loaded_config[0]}
+        http_port=${loaded_config[1]}
+        use_same=${loaded_config[2]}
+        use_auth=${loaded_config[3]}
+        username=${loaded_config[4]}
+        password=${loaded_config[5]}
+        https_host=${loaded_config[6]}
+        https_port=${loaded_config[7]}
+        ftp_host=${loaded_config[8]}
+        ftp_port=${loaded_config[9]}
+        choice="set"
 fi
 
-if [ "$choice" = "set" ]; then
-
-    # default values
-    http_host=""
-    http_port=""
-    use_same="no"
-    https_host=""
-    https_port=""
-    ftp_host=""
-    ftp_port=""
-    username=""
-    password=""
-
-    echo
-    echo -e " \e[4mEnter details \e[0m : \e[2m\e[3m (leave blank if you don't to use any proxy settings) \e[0m"
-    echo
-    echo -ne "\e[36m HTTP Proxy host \e[0m"; read http_host
-    echo -ne "\e[32m HTTP Proxy port \e[0m"; read http_port
-    echo -ne "\e[0m Use same for HTTPS and FTP (y/n) ? \e[0m"; read use_same
-    echo -ne "\e[0m Use authentication (y/n) ? \e[0m        "; read use_auth
-
-    if [[ "$use_auth" = "y" || "$use_auth" = "Y" ]]; then
-        read -p " Enter username                 : " username
-        echo -n " Enter password (use %40 for @) : " ; read -s password
+if [ "$1" != "load" ]; then
+    echo -e "
+    \e[1m\e[33mProxyMan
+    =========\e[0m
+    Tool to set up system wide proxy settings on Linux.
+    \e[2m\e[33m🌟\e[0m\e[3m Star it \e[0m : \e[4m\e[34m https://github.com/himanshub16/ProxyMan \e[0m
+
+    \e[4mThe following options are available : \e[0m
+    \e[1m set \e[0m    : \e[2m Set proxy settings \e[0m
+    \e[1m unset \e[0m  : \e[2m Unset proxy settings \e[0m
+    \e[1m list \e[0m   : \e[2m List current settings \e[0m
+
+    "
```

Comment:
These lines are indented, and thus appear the same when the script runs. They should be aligned to the left.

Reply:
Done 👍